staging/src/k8s.io/controller-manager/pkg/leadermigration/options/options_test.go

```diff
@@ -99,6 +99,67 @@ controllerLeaders: []
 				ControllerLeaders: []config.ControllerLeaderConfiguration{},
 			},
 		},
+		{
+			name:              "enabled, with custom configuration file (version v1)",
+			flags:             []string{"--enable-leader-migration"},
+			enableFeatureGate: true,
+			expectEnabled:     true,
+			configContent: `
+apiVersion: controllermanager.config.k8s.io/v1
+kind: LeaderMigrationConfiguration
+leaderName: test-leader-migration
+controllerLeaders: []
+`,
+			expectErr: false,
+			expectConfig: &config.LeaderMigrationConfiguration{
+				LeaderName:        "test-leader-migration",
+				ResourceLock:      "leases",
+				ControllerLeaders: []config.ControllerLeaderConfiguration{},
+			},
+		},
+		{
+			name:              "enabled, with populated controllerLeaders (version v1)",
+			flags:             []string{"--enable-leader-migration"},
+			enableFeatureGate: true,
+			expectEnabled:     true,
+			configContent: `
+apiVersion: controllermanager.config.k8s.io/v1
+kind: LeaderMigrationConfiguration
+leaderName: test-leader-migration
+controllerLeaders:
+  - name: route
+    component: "*"
+  - name: service
+    component: "*"
+  - name: cloud-node-lifecycle
+    component: "*"
+  - name: nodeipam
+    component: "*"
+`,
+			expectErr: false,
+			expectConfig: &config.LeaderMigrationConfiguration{
+				LeaderName:   "test-leader-migration",
+				ResourceLock: "leases",
+				ControllerLeaders: []config.ControllerLeaderConfiguration{
+					{
+						Name:      "route",
+						Component: "*",
```

Comment:
Can we use a non star component? I believe the default is "*" and I would like to better test this use case which would have either "kube-controller-manager" or "cloud-controller-manager"?

Reply:
added a new test case for that.